src/components/homepage/HomepageList.js

```diff
@@ -0,0 +1,34 @@
+import React, { useEffect } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import { useNavigate } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+import { getCompanies } from '../../redux/companies-list';
+import './HomepageList.css';
+import HomepageItem from './HomepageItem';
+
+const HomepageList = () => {
+  const navigate = useNavigate();
+  const { companyList } = useSelector((state) => state.companies);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(getCompanies());
+  }, [dispatch]);
+
+  return (
+    <div className="context">
+      <div className="container">
+        {companyList.map((company) => (
+          <div
+            key={uuidv4()}
+            onClick={() => navigate(`/details/${company.symbol}`)}
+            aria-hidden="true"
+          >
+            <HomepageItem title={company.symbol} />
+          </div>
+        ))}
+      </div>
+    </div>
+  );
+};
+
```

Comment:
- [ ] Based on the requirements there should be a search or filter mechanism on the home page that allows the users to filter items on the home page. Kindly implement that.

  ![furkan-filter](https://user-images.githubusercontent.com/36542986/197741201-6b844180-5826-4f47-bb03-150713dc2b5e.jpg)

Reply:
- [ ] As per the previous reviewer kindly add a search or filter mechanism on the home page that allows the users to filter items on the home page. Kindly implement that.